lib/shopify-cli/monorail/log.rb

```diff
@@ -0,0 +1,77 @@
+require 'json'
+require 'securerandom'
+require 'time'
+require 'timeout'
+
+module ShopifyCli
+  module Monorail
+    class Log
+      INVOCATIONS_SCHEMA = 'app_cli_command/1.0'
+
+      SUCCESS_SENTINEL = '_success'
+
+      MICROSECOND_PRECISION = 6
+
+      def initialize(writable:)
+        @writable = writable
+        @detail_ids = [] # stack of detail ids used to keep track of nested calls
+      end
+
+      def invocation(name, args, &block)
+        args.unshift(name)
+        payload = {
+          cli_sha: ShopifyCli::Helpers::Git.sha(dir: ShopifyCli::ROOT),
```

Comment:
We already do this in the API

```
    def current_sha
      output, status = ctx.capture2e('git', 'rev-parse', 'HEAD', chdir: ShopifyCli::ROOT)
      status.success? ? output.strip : 'SHA unavailable'
    end
```

Should we update the API to use this git helper as well?

Reply:
Yeah, I think we should isolate and reuse stuff like this as much as possible. I'll update.